testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/stateless/callerprincipal/TestEJB2Bean.java

```
@@ -0,0 +1,41 @@
+package org.jboss.as.test.integration.ejb.stateless.callerprincipal;
+
+import javax.ejb.CreateException;
+import javax.ejb.SessionBean;
+import javax.ejb.SessionContext;
+import java.rmi.RemoteException;
+import java.util.logging.Logger;
+
+public class TestEJB2Bean implements SessionBean, TestEJB2Int {
+
+    private static final Logger log = Logger.getLogger(TestEJB2Bean.class.getName());
+
+    private static final String USER = "user1";
+
+    public boolean isCallerInRole(String role) {
+        String caller = ctx.getCallerPrincipal().getName();
+        if (!USER.equals(caller)) {
+            log.severe("Caller name is not " + USER + ", but " + caller);
```

Should this fail instead of just logging?